File: src/main/java/com/braintreegateway/SearchCriteria.java
package com.braintreegateway;

import java.util.List;

public class SearchCriteria extends BaseRequest {
    private String xml;

    public SearchCriteria(String type, Object value) {
        this.xml = RequestBuilder.buildXMLElement(type, value);
    }

    public SearchCriteria(List<?> items) {
        StringBuilder builder = new StringBuilder();
        for (Object item : items) {
            builder.append(RequestBuilder.buildXMLElement("item", item.toString()));
        }
        this.xml = builder.toString();
    }

    @Override
    public String toXML() {
        return this.xml;
    }

    @Override
    public String toQueryString(String parent) {
        throw new UnsupportedOperationException();
    }

    @Override
    public String toQueryString() {
        throw new UnsupportedOperationException();
    }
}

File: src/main/scala/com/braintreegateway/search/SearchCriteria.scala
package com.braintreegateway.search

import com.braintreegateway.BaseRequest
import com.braintreegateway.RequestBuilder

class SearchCriteria private(val xml: String) extends BaseRequest {

  def this(items: List[_]) = {
    this(items.foldLeft(new StringBuilder) {
      (buf, item) => buf.append(
        RequestBuilder.buildXmlElementString("item", item.toString)
      )
    }.toString)
  }

  def this(searchType: String, value: AnyRef) = {
    this(RequestBuilder.buildXmlElementString(searchType, value))
  }

  override def toXmlString = xml

  override def toQueryString(parent: String): String = {
    throw new UnsupportedOperationException
  }

  override def toQueryString: String = {
    throw new UnsupportedOperationException
  }
}